pkg/resource/apply.go

```diff
@@ -18,70 +18,74 @@ package resource
 
 import (
 	"bytes"
+	"io"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer/json"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/cli-runtime/pkg/genericclioptions/printers"
 	kcmd "k8s.io/kubernetes/pkg/kubectl/cmd"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 )
 
-// Apply applies the given resource bytes to the target cluster specified by kubeconfig
-func (r *Helper) Apply(obj []byte) error {
+type ApplyResult string
+
+var (
+	ConfiguredApplyResult ApplyResult = "configured"
+	UnchangedApplyResult  ApplyResult = "unchanged"
+	CreatedApplyResult    ApplyResult = "created"
+	UnknownApplyResult    ApplyResult = "unknown"
+)
```

Comment:
This is fantastic, I actually had a Jira card to add this someday.

Reply:
It led me to find out that we were always patching :)